### PR TITLE
Beggars 2: implement actual begging and begin to add a simple storyline

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
```SUMMARY: Content "Adds begging and begins adding a storyline to refugee center beggars"```

#### Purpose of change

I implemented the Beggars as a very rudimentary set of static NPCs. We now have the tools to make them functional.

#### Describe the solution

Using bulk items, the Beggars accept food donations.
When they've had something to eat, if you have a base camp, it's possible to try to convince them to come to it. This gets easier the more you've worked to earn their trust, whether by food donations or doing other nice things for them.
They will only come with you if you've gotten all of them to agree to it.

Dino Dave is an exception. He gets his own story.

#### Additional context

In the process I will migrate the beggars into their own folder and get their files to match the style I used for the refugee center NPCs, which is a bit easier to read.

### CODE REVIEW

File: data/json/npcs/beggars/BEGGAR_3_Luo_Meizhen.json [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+        "u_has_var": "luo_recruited",
+        "yes": "Yo.  Anyone else keen on moving from this bus stop to your tent city?",
+        "no": {
+          "npc_has_var": "dont_insult_luo",
+          "yes": {
+            "npc_has_var": "apology_luo",
+            "yes": {
+              "npc_has_effect": "beggar_has_eaten",
+              "yes": "Hey there.  Good to see you again.",
+              "no": "Careful, I'm getting hangry again and am not totally responsible for my own actions."
```
Comment (Night-Pryanik) [inline data/json/npcs/beggars/BEGGAR_3_Luo_Meizhen.json]: Hungry or angry?

File: data/json/npcs/beggars/BEGGAR_3_Luo_Meizhen.json [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+        "topic": "TALK_REFUGEE_BEGGAR_3_COMPLAIN"
+      },
+      {
+        "condition": { "npc_has_effect": "beggar_has_eaten" },
+        "text": "What's up?",
+        "topic": "TALK_REFUGEE_BEGGAR_3_INTRO"
+      },
+      {
+        "condition": { "not": { "npc_has_effect": "beggar_has_eaten" } },
+        "text": "I might have some food for you.  Are you hungry?.",
```
Comment (Night-Pryanik) [inline data/json/npcs/beggars/BEGGAR_3_Luo_Meizhen.json]:
```suggestion
        "text": "I might have some food for you.  Are you hungry?",
```

File: data/json/npcs/beggars/BEGGAR_3_Luo_Meizhen.json [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        "text": "I have a camp of my own, away from here.  No paperwork required.  Want to come?",
+        "topic": "TALK_REFUGEE_BEGGAR_3_RECRUIT1"
+      },
+      { "text": "I'd better get going.", "topic": "TALK_DONE" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_REFUGEE_BEGGAR_3_RACISM",
+    "dynamic_line": "Sure. My grandparents were from China. That means I'm obviously personally responsible for all this.  Do you think there's some other reason they let hundreds of other educated people in and I'm sitting out here?",
```
Comment (Night-Pryanik) [inline data/json/npcs/beggars/BEGGAR_3_Luo_Meizhen.json]:
```suggestion
    "dynamic_line": "Sure.  My grandparents were from China.  That means I'm obviously personally responsible for all this.  Do you think there's some other reason they let hundreds of other educated people in and I'm sitting out here?",
```

File: data/json/npcs/beggars/BEGGAR_3_Luo_Meizhen.json [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+    "type": "talk_topic",
+    "id": "TALK_REFUGEE_BEGGAR_3_JOIN",
+    "dynamic_line": "That's awful kind of you, but look at me.  I'm not travelling material, I've managed to stay fifty pounds overweight on a diet of pine nuts and wilted rhubarb, and I scream and shake uncontrollably at the sight of blood.",
+    "responses": [
+      { "text": "Okay, yeah, that's a bit of a problem.  What were you saying before?", "topic": "TALK_NONE" },
+      { "text": "Why stay out here then?", "topic": "TALK_REFUGEE_BEGGAR_3_WHYSTAY" },
+      { "text": "What did you do before <the_cataclysm>?", "topic": "TALK_REFUGEE_BEGGAR_3_EXPERTISE" },
+      {
+        "condition": { "and": [ "u_has_camp", { "not": { "u_has_var": "luo_recruited" } } ] },
+        "text": "It'd be temporary. I have a base set up. There are only a few of us survivors left, we need to work together",
```
Comment (Night-Pryanik) [inline data/json/npcs/beggars/BEGGAR_3_Luo_Meizhen.json]:
```suggestion
        "text": "It'd be temporary.  I have a base set up.  There are only a few of us survivors left, we need to work together.",
```

File: data/json/npcs/beggars/BEGGAR_1_Reena_Sandhu.json [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+    "id": "TALK_REFUGEE_BEGGAR_1_INTRO",
+    "dynamic_line": {
+      "npc_has_effect": "beggar_has_eaten",
+      "yes": "They won't let me in.  They say they're too full.  I'm allowed to camp out here as long as I keep it clean and don't make a fuss, but I'm reduced to begging to survive.",
+      "no": "They won't let me in.  They say they're too full.  I'm allowed to camp out here as long as I keep it clean and don't make a fuss, but I'm so hungry."
+    },
+    "responses": [
+      { "text": "Why don't you go somewhere else?", "topic": "TALK_REFUGEE_BEGGAR_1_LEAVE" },
+      { "text": "Why don't you scavenge your own food?", "topic": "TALK_REFUGEE_BEGGAR_1_SCAVENGE" },
+      { "text": "What did you do before the cataclysm?", "topic": "TALK_REFUGEE_BEGGAR_1_EXPERTISE" },
```
Comment (Night-Pryanik) [inline data/json/npcs/beggars/BEGGAR_1_Reena_Sandhu.json]:
```suggestion
      { "text": "What did you do before <the_cataclysm>?", "topic": "TALK_REFUGEE_BEGGAR_1_EXPERTISE" },
```

File: data/json/npcs/beggars/BEGGAR_1_Reena_Sandhu.json [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+      { "text": "No problem.  See you around.", "topic": "TALK_DONE", "effect": "u_bulk_trade_accept" }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_REFUGEE_BEGGAR_1_LEAVE",
+    "dynamic_line": "Where else?  I can't fight those things out there.  I'm in terrible physical condition, don't have any useful skills, and I'm terrified of <zombies> and violence.  How am I supposed to find a safe place?",
+    "responses": [
+      { "text": "Why don't you scavenge your own food?", "topic": "TALK_REFUGEE_BEGGAR_1_SCAVENGE" },
+      { "text": "What did you do before the cataclysm?", "topic": "TALK_REFUGEE_BEGGAR_1_EXPERTISE" },
```
Comment (Night-Pryanik) [inline data/json/npcs/beggars/BEGGAR_1_Reena_Sandhu.json]:
```suggestion
      { "text": "What did you do before <the_cataclysm>?", "topic": "TALK_REFUGEE_BEGGAR_1_EXPERTISE" },
```

File: data/json/npcs/beggars/BEGGAR_1_Reena_Sandhu.json [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+      }
+    ]
+  },
+  {
+    "type": "talk_topic",
+    "id": "TALK_REFUGEE_BEGGAR_1_SCAVENGE",
+    "dynamic_line": "Out there?  That's suicide!  People that go out there don't come back, people who can hold their own... unlike me.  I'd rather take my chances begging for scraps and waiting for someone in the center to die and make room for me, thanks.",
+    "responses": [
+      { "text": "Why don't you go somewhere else?", "topic": "TALK_REFUGEE_BEGGAR_1_LEAVE" },
+      { "text": "What did you do before the cataclysm?", "topic": "TALK_REFUGEE_BEGGAR_1_EXPERTISE" },
```
Comment (Night-Pryanik) [inline data/json/npcs/beggars/BEGGAR_1_Reena_Sandhu.json]:
```suggestion
      { "text": "What did you do before <the_cataclysm>?", "topic": "TALK_REFUGEE_BEGGAR_1_EXPERTISE" },
```

File: data/json/npcs/beggars/BEGGAR_3_Luo_Meizhen.json (hunk 1 above)
Comment (I-am-Erk) [reply data/json/npcs/beggars/BEGGAR_3_Luo_Meizhen.json]: "hangry" is a slang word that means angry because you're hungry

**Comment by I-am-Erk** [comment]
OK. Getting long, so I have cut out changes to beggars 4 and 5 for now, and beggar 2 was always going to be his own PR. We'll see what gorgon and travis think, this should be about ready now.

I'm sure I've managed to get some dialogue tags wrong, although I did run a few times through playtest. I also will need playtesters to ensure that recruiting Reena is really possible.

**Pull request merged**